src/analysis/special_functions/exp_log.lean

```diff
@@ -282,38 +282,99 @@ namespace real
 
 variables {x y z : ℝ}
 
-lemma exists_exp_eq_of_pos {x : ℝ} (hx : 0 < x) : ∃ y, exp y = x :=
-have ∀ {z:ℝ}, 1 ≤ z → z ∈ set.range exp,
-  from λ z hz, intermediate_value_univ 0 (z - 1) continuous_exp
-    ⟨by simpa, by simpa using add_one_le_exp_of_nonneg (sub_nonneg.2 hz)⟩,
-match le_total x 1 with
-| (or.inl hx1) := let ⟨y, hy⟩ := this (one_le_inv hx hx1) in
-  ⟨-y, by rw [exp_neg, hy, inv_inv']⟩
-| (or.inr hx1) := this hx1
+/-- The real exponential function tends to `+∞` at `+∞`. -/
+lemma tendsto_exp_at_top : tendsto exp at_top at_top :=
+begin
+  have A : tendsto (λx:ℝ, x + 1) at_top at_top :=
+    tendsto_at_top_add_const_right at_top 1 tendsto_id,
+  have B : ∀ᶠ x in at_top, x + 1 ≤ exp x :=
+    eventually_at_top.2 ⟨0, λx hx, add_one_le_exp_of_nonneg hx⟩,
+  exact tendsto_at_top_mono' at_top B A
 end
 
+/-- The real exponential function tends to `0` at `-∞` or, equivalently, `exp(-x)` tends to `0`
+at `+∞` -/
+lemma tendsto_exp_neg_at_top_nhds_0 : tendsto (λx, exp (-x)) at_top (𝓝 0) :=
+(tendsto_inv_at_top_zero.comp tendsto_exp_at_top).congr (λx, (exp_neg x).symm)
+
+/-- The real exponential function tends to `1` at `0`. -/
+lemma tendsto_exp_nhds_0_nhds_1 : tendsto exp (𝓝 0) (𝓝 1) :=
+by { convert continuous_exp.tendsto 0, simp }
+
+lemma tendsto_exp_at_bot : tendsto exp at_bot (𝓝 0) :=
+(tendsto_exp_neg_at_top_nhds_0.comp tendsto_neg_at_bot_at_top).congr $
+  λ x, congr_arg exp $ neg_neg x
+
+lemma tendsto_exp_at_bot_nhds_within : tendsto exp at_bot (𝓝[set.Ioi 0] 0) :=
+tendsto_inf.2 ⟨tendsto_exp_at_bot, tendsto_principal.2 $ eventually_of_forall exp_pos⟩
+
+lemma range_exp : set.range exp = set.Ioi 0 :=
+set.ext $ λ y, ⟨λ ⟨x, hx⟩, hx ▸ exp_pos x,
+  λ hy, mem_range_of_exists_le_of_exists_ge continuous_exp
+    (tendsto_exp_at_bot.eventually (ge_mem_nhds hy)).exists
+    (tendsto_exp_at_top.eventually (eventually_ge_at_top y)).exists⟩
```

Comment:
Could you deduce this from general theory too?  I wrote `surjective_of_continuous` at some point (in `topology.algebra.ordered`), could deduce directly from this or write a relative version like
```
lemma surjective_of_continuous_Ioo {a : ℝ} {f : ℝ → ℝ} (hf' : ∀ x, f x ∈ set.Ioi a)
  (hf : continuous f) (h_top : tendsto f at_top at_top) (h_bot : tendsto f at_bot (𝓝[set.Ioi a] a)) :
  set.range f = set.Ioi a :=
```

Reply:
I think that it's easier to reuse `mem_range_of_exists_le_of_exists_ge` than define lots of lemmas like `surjective_of_continuous_*`.